music_assistant/providers/radioparadise/__init__.py

```diff
@@ -0,0 +1,369 @@
+
+        # Fetch the current metadata to get the latest cover image for the browse view.
+        metadata = await self._get_channel_metadata(channel_id)
+        cover_url = cast("str | None", metadata.get("cover_url")) if metadata else None
+
+        if cover_url:
+            images = [
+                MediaItemImage(
+                    provider=self.lookup_key,
+                    type=ImageType.THUMB,
+                    path=cover_url,
+                    remotely_accessible=True,
+                )
+            ]
+            radio.metadata.images = UniqueList(images)
+
+        return radio
+
+    async def _get_channel_metadata(self, channel_id: str) -> dict[str, Any] | None:
+        """Get current playing metadata for a channel."""
+        if channel_id not in self._channels_cache:
+            return None
+
+        channel_info = self._channels_cache[channel_id]
+        api_url = channel_info["api_url"]
+
+        try:
+            timeout = aiohttp.ClientTimeout(total=10)
+            async with (
+                aiohttp.ClientSession(timeout=timeout) as session,
+                session.get(api_url) as response,
+            ):
+                if response.status != 200:
+                    self.logger.debug(f"API call to {api_url} failed with status {response.status}")
+                    return None
+
+                data = await response.json()
+                if "artist" in data:
+                    current_song = data
+                elif "song" in data and len(data["song"]) > 0:
+                    current_song = data["song"][0]
+                else:
+                    self.logger.debug(f"No song data in API response for channel {channel_id}")
+                    return None
+
+                return {
+                    "title": current_song.get("title", ""),
+                    "artist": current_song.get("artist", ""),
+                    "cover_url": current_song.get("cover"),
+                    "duration": current_song.get("time"),
+                }
+        except Exception as exc:
+            self.logger.debug(f"Failed to get metadata for channel {channel_id}: {exc}")
+            return None
+
+    def _build_stream_url(self, channel_id: str) -> str:
+        """Build stream URL for a channel with fallback to other formats."""
+        if channel_id not in self._channels_cache:
+            return ""
+
+        channel_info = self._channels_cache[channel_id]
+        stream_urls = channel_info.get("stream_urls", {})
+
+        current_format_index = -1
+        with contextlib.suppress(ValueError):
+            current_format_index = FALLBACK_ORDER.index(self._stream_format)
+
+        for format_key in FALLBACK_ORDER[current_format_index:]:
+            if format_key in stream_urls:
+                return cast("str", stream_urls[format_key])
+
+        return ""
+
+    async def _monitor_stream_metadata(self, item_id: str) -> None:
```

Comment:
So, I would instead pass the streamdetails to this method and store the task in the streamdetails.data 
in the "on_streamed" callback you can cancel the task and clean it up.

I'll make sure that we extend the streamtitle to streammetadata so we can pass on more live details